micropython/upysh/upysh.py

```diff
@@ -53,10 +52,23 @@ def head(f, n=10):
                 break
             sys.stdout.write(l)
 
-
 def cat(f):
     head(f, 1 << 30)
 
+def cp(s, t):
+    try: 
+        if os.stat(t)[0] & 0x4000:  # is directory
+            t = t.rstrip("/") + "/" + s
+    except OSError:
+        pass
+    buf = bytearray(512)
+    buf_mv = memoryview(buf)
+    with open(s, "rb") as s:
+        with open(t, "wb") as t:
+            while True:
+                n = s.readinto(buf)
+                if n <= 0: break
+                t.write(buf_mv[:n])
```

Comment:
This could be simpler, like this:
```python
with open(s, "rb") as s, open(t, "wb") as t:
    buf = memoryview(bytearray(512))
    while n := s.readinto(buf) > 0:
        t.write(buf[:n])
```

Reply:
Yes. But the assignment operator is not available in all MicroPython variants. So I kept it clumsy.